azure-proto-core/DeletableResourceOperations.cs

```diff
@@ -0,0 +1,26 @@
+using Azure;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace azure_proto_core
+{
+    /// <summary>
+    /// Implementation for resources that implement the immutable resource pattern
+    /// </summary>
+    public abstract class DeletableResourceOperations<TResource, TOperations> : GenericResourcesOperations <TResource, TOperations>
```

Comment:
We should be consistent in naming (GenericResourceOperations, DeletableResourceOperations) or (GenericResourcesOperations, DeletableResourcesOperations)

Reply:
Just a thought - alternately, what would it look like to have interfaces for Create and Delete operations, rather than a class hierarchy?  In this case, we would have ResourcesOperations, and separate interfaces for Create, Delete, and Tagging operations?  The downside is that any change in the operations would be a breaking change, but the advantage would be that they can be applied in any pattern to a resource.

Reply:
> We should be consistent in naming (GenericResourceOperations, DeletableResourceOperations) or (GenericResourcesOperations, DeletableResourcesOperations)

I suggest using Resource Singular for both